juju/controller.py

```diff
@@ -257,13 +263,20 @@ def logout(self, force=False):
         """
         pass
 
-    def get_model(self, name):
+    async def get_model(self, uuid):
         """Get a model by name.
 
         :param str name: Model name
 
         """
-        pass
+        model_facade = client.ModelManagerFacade()
+        model_facade.connect(self.connection)
+        results = await model_facade.ModelInfo([client.Entity(tag.model(uuid))])
+        if results and results.results and results.results[0].result:
+            return results.results[0].result
+        else:
+            log.debug('No result found for model: ', uuid)
+            return None
```

Comment:
I'd be tempted to raise an errors.JujuError here. This isn't a "model.exists" call, but a "get me the model!" command, and I think that it is only under Exceptional circumstances that this would come back empty.

Bonus points for writing a NotFoundError, and raising that instead :-)